monitoring/perf_context_imp.h

```diff
@@ -59,6 +59,20 @@ extern __thread PerfContext perf_context;
     perf_context.metric += value;              \
   }
 
+// Increase metric value
+#define PERF_COUNTER_BY_LEVEL_ADD(metric, value, level)                      \
+  if (perf_level >= PerfLevel::kEnableCount &&                               \
+      perf_context.per_level_perf_context_enabled &&                         \
+      perf_context.level_to_perf_context) {                                  \
+    if ((*(perf_context.level_to_perf_context))[level]) {                    \
```

Comment:
I read std::map doc, but determining non existing element to be null and set it if it is still confusing to me: https://en.cppreference.com/w/cpp/container/map/operator_at

I suggest we write it in a more explicit way: find() and if we can't find it, set a new one.